wallet-user/js-example/index.js

```diff
@@ -1,83 +1,41 @@
-const ethers = require('ethers');
-
-// Replace with your private key (be cautious with private keys!)
-const privateKey = 'YOUR_PRIVATE_KEY';
-// replace with your ERC20 receiver
-const toAddress = 'RECIPIENT_ADDRESS';
-// ERC20 token contract address (replace with the address of the token you want to send)
-const tokenAddress = 'TOKEN_CONTRACT_ADDRESS';
-
-const web3ProviderEndpoint = 'https://bsc-dataseed.bnbchain.org';
-const paymasterEndpoint = 'https://bsc-megafuel.nodereal.io';
-
-// testnet endpoint
-// const web3ProviderEndpoint = 'https://bsc-testnet-dataseed.bnbchain.org';
-// const paymasterEndpoint = 'https://bsc-megafuel-testnet.nodereal.io';
-
-
-class PaymasterProvider extends ethers.providers.JsonRpcProvider {
-  constructor(url) {
-    super(url);
-  }
-
-  async isSponsorable(transaction) {
-    const params = [{
-      to: transaction.to,
-      from: transaction.from,
-      value: transaction.value != null ? ethers.utils.hexlify(transaction.value) : '0x0',
-      gas: ethers.utils.hexlify(transaction.gasLimit || 0),
-      data: transaction.data || '0x'
-    }];
-
-    const result = await this.send('pm_isSponsorable', params);
-    return result;
-  }
-}
+import 'dotenv/config';
+import {ethers} from "ethers";
+import { PaymasterClient } from 'megafuel-js-sdk';
 
 async function sendERC20Transaction() {
-
-  // Provider for assembling the transaction (e.g., mainnet)
-  const assemblyProvider = new ethers.providers.JsonRpcProvider(web3ProviderEndpoint);
-
   // Provider for sending the transaction (e.g., could be a different network or provider)
-  const paymasterProvider = new PaymasterProvider(paymasterEndpoint);
-
-  const wallet = new ethers.Wallet(privateKey, assemblyProvider);
+  const paymasterProvider = new PaymasterClient(process.env.PAYMASTER_URL);
```

Comment:
It should be paymasterClient instead of paymasterProvider

Reply:
fixed